### PR TITLE
Feature/170 product description subnav

### PR DESCRIPTION
**Pull request by uditdesai**
## Description
**TLDR:** This PR updates the product description for courses to feature a new sticky header consisting of jump links to the appropriate sections in the description. It was quite finicky so added some implementation details below:

### How I built it
I check if it's a course or not using the same logic present in other code snippets checking whether the bookings metafield exists and if its type is 'Course' or not. If the type is not 'Course', the regular Shopify product description from the admin is used. If it is 'Course', it uses fields from Accentuate.

#### Accentuate fields added
I added a repeatable section with two fields within it:
1. title -> this shows up on the sticky header and as the main title of the section
2. markdown -> the markdown content used within the section

I was also going to add an id field used for the jump links, however, I opted to use the `handleize` Liquid string filter which turns the title into a handle.

#### Jump links
For the jump links, I just used an `a` tag that links to the `id` so it doesn't require JS but additionally appends to the URL which is an added benefit.

#### Active state
To keep track of active states via scroll, I used JS to track which section is on the screen and update the active state accordingly. This added some complexity where I had to do three separate cases since the first section and last section's active state should remain turned on before/after scrolling through it (i.e. The first tab should be active on page load and as you scroll down to it).

#### UX
Scrolling to the section on click and scrolling through sections got quite finicky and complex due to having two sticky headers, having the sticky tab header be sometimes horizontally scrollable, and different heights at different screen sizes.

As a result, I had to add some _imaginary_ spacing on top of sections so when the window scrolls to the section, the title actually appears below the two sticky headers. However, since the height of both headers change, I have to keep track of this using JS and media queries in CSS.

One solution would have been using the CSS property `scroll-margin-top`, but it doesn't work in safari :((

Additionally, Jake wanted smooth scrolling which works great in Chrome, but for Safari, it would require a JS implementation which I did think was worth doing. Open to doing it if it's a necessity though. 

**See Picture here**
Screen Shot 2021-05-18 at 3 50 36 PM

## Type(s) of changes

- [X] New feature

## Motivation for PR

Issue: https://github.com/sanctuarycomputer/nunu/issues/170

## How Has This Been Tested?

I tested this with different screen sizes and browsers to ensure scrolling works as intended. 

Some things to look out for and test:
- Making the sticky nav horizontally scrollable (happens at certain screen sizes)
- Testing on Safari
- I used the Markdown snippet for markdown but certain things might look off if this snippet was intended for a different part of the site?

Preview: https://ch9myxgw61p9oj0r-52674822324.shopifypreview.com
Main product with data added on Accentuate: https://index-space.org/products/animation-overview
Second product with data added on Accentuate: https://index-space.org/products/intro-to-coding-for-designers

## Other Links:

Twist thread: https://twist.com/a/133876/ch/376069/t/2423429/c/67189549
Figma: https://www.figma.com/file/o8XuPfORWc9bH7tB20vImN/Index-Web-%F0%9F%9F%A2?node-id=2093%3A2808

## Applicable screenshots:
Screen Shot 2021-05-18 at 4 03 05 PM

### CODE REVIEW

File: theme/scripts/productDescription.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+export default(function() {
+	const ProductDescription = {
```
Comment (sepowitz) [inline theme/scripts/productDescription.ts:2]: Some strange formatting here, any chance we can format it a bit nicer? (doesn't look like we have prettier set up on this project, but if you are using VSCode or similar they have a local plugin which can handle it)

File: theme/scripts/productDescription.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		SELECTORS: {
+			tab: "[data-product-description-tab]",
+      content: "[data-product-description-content]",
+      tabsWrapper: ".ProductDescription__tabs",
```
Comment (sepowitz) [inline theme/scripts/productDescription.ts:6]: We're generally trying to stick to using `data-attributes` for handling DOM in JS. Allows us to distinguish from presentational attributes like classes. The activeTab class is cool though, although you could also use a data-attribute if you wanted.

File: theme/scripts/productDescription.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        });
+      })
+		},
+
+    getStickyHeight(tabsWrapper: HTMLElement, navs: HTMLElement[]): number {
+      const tabsWrapperHeight = tabsWrapper.offsetHeight;
+      const activeNav = navs.find((elem) => {
+        if (elem.offsetHeight !== 0) return elem;
+      })
+      return tabsWrapperHeight + activeNav.offsetHeight + 16;
```
Comment (sepowitz) [inline theme/scripts/productDescription.ts:47]: What's `16` in this case? We generally like to store these type of loose integers in a named constant so that it's a bit clearer what it represents. You can declare this outside the module, and we like to use all caps e.g. `const STICKY_HEIGHT_BUFFER = 16;`

File: theme/scripts/productDescription.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      } else {
+        tab.classList.remove(
+          ProductDescription.CLASSES.activeTab
+        );
+      }
+    },
+
+		scrollHandler(tabs: HTMLElement[], tabbedContent: HTMLElement[], height: number) {
+      tabbedContent.forEach((elem, index) => {
+        const offsetTop = elem.offsetTop;
```
Comment (sepowitz) [inline theme/scripts/productDescription.ts:64]: Are we restricting tabbedContent to only 3? If so, we should probably ensure some type of validation at the accentuate level.

File: theme/styles/snippets/product-description.scss [hunk 5]
```diff
@@ -0,0 +1,84 @@
+.ProductDescription {
+  padding-bottom: 3.125rem;
+
+  h1 {
```
Comment (sepowitz) [inline theme/styles/snippets/product-description.scss:4]: We generally prefer classes over referencing elements by name. This way we can easily change the element without having to worry about breaking styles :)

**Review by sepowitz — COMMENTED** [review]
@uditdesai This is really great, TY! Added a few comments, mostly style related, as well as a couple questions. Let me know if you have any questions 💯

File: theme/scripts/productDescription.ts (hunk 1 above)
Comment (uditdesai) [reply theme/scripts/productDescription.ts:2]: I have Prettier setup but my settings must be different than what the repo uses (2 spaces vs 4 spaces, etc), I'll try to adjust this

File: theme/scripts/productDescription.ts (hunk 3 above)
Comment (uditdesai) [reply theme/scripts/productDescription.ts:47]: the 16 is just some sort of buffer that ended up having the best experience in terms of when the active state switches from one tab to the other. I chose 16 just because it's 1 rem. It's a bit finicky because I'm sure any number between 10-20px would work fine. 

I'll make it a constant and add a comment.

File: theme/scripts/productDescription.ts (hunk 4 above)
Comment (uditdesai) [reply theme/scripts/productDescription.ts:64]: Is there something in the code that makes it seem like only three? I kept it flexible so it could be as many tabs as Elie wants.

**Short answer: Can be more than 3**, but I never thought about the case where there is only one and would be no need for a sticky header? I'll ask Jake.

Comment (chrismekim) [reply theme/scripts/productDescription.ts:64]: @sepowitz when I talked to Jake, he mentioned the number of tabs should be flexible in case they want more than 3. BUT it doesn't look great when the nav content gets really long and needs to side scroll. 🤷🏻‍♀️ What do you think @uditdesai?

**Review by chrismekim — APPROVED** [review]

File: theme/scripts/productDescription.ts (hunk 4 above)
Comment (uditdesai) [reply theme/scripts/productDescription.ts:64]: Agreed that the default chrome scrollbar horizontally doesn't look great. But, for smaller screen sizes, it's going to have to happen. Maybe keep the font size smaller and set a max to avoid the scroll for most screen sizes?

File: theme/scripts/productDescription.ts (hunk 2 above)
Comment (uditdesai) [reply theme/scripts/productDescription.ts:6]: Cool that makes sense!

File: theme/scripts/productDescription.ts (hunk 4 above)
Comment (sepowitz) [reply theme/scripts/productDescription.ts:64]: Apologies! I misread the `scrollHandler` method. Let's hide the scroll bar with css, and ask Jake for notes in QA.

**Comment by uditdesai** [comment]
Made edits according to comments and fixed the file formatting! @sepowitz

**Review by sepowitz — APPROVED** [review]
Gorgeous! Thanks so much @uditdesai feel free to merge
